### DISCUSSION TITLE
trying to bind an oracle numeric column to a string value gives:AttributeError: 'str' object has no attribute 'is_infinite'

### DISCUSSION THREAD

**Discussion by dvht** [sqlalchemy/sqlalchemy#9876] [Usage Questions] [answered]
Oracle, is able to do implicit conversion.  So passing a string value to a numeric column works.  Workaround is of course pass the right type to the column.

But if we change this part :
https://github.com/sqlalchemy/sqlalchemy/blob/d9d0ffd96c632750be9adcb03a207d75aecaa80f/lib/sqlalchemy/dialects/oracle/cx_oracle.py#L484
```
            def process(value):
                if isinstance(value, (int, float)):
                    return processor(value)
                elif value is not None and value.is_infinite():    # only succeeds with value=None or Decimal
                    return float(value)
                else:
                    return value

```
and return the string 
```
            def process(value):
                if isinstance(value, (int, float)):
                    return processor(value)
                elif value is not None and hasattr(value,'is_infinite') and value.is_infinite():
                    return float(value)
                else:
                    return value

```
I don't see how this would break anything

**Comment by zzzeek** [comment]
hi -

can you please describe the complete context for the problem you are seeking to solve, thanks.  I dont know what it is you're referring towards

**Reply by zzzeek** [reply, marked as answer]
passing a string to a numeric....we don't support that.  please pass Decimal objects.  if you want to pass strings then you can [create a custom type](https://docs.sqlalchemy.org/en/20/core/custom_types.html#types-custom)

please note it's not as simple as "change this line of code".  that line of code will get changed back someday, unless we add new tests and then fully support this case.  then if someday cx_oracle or oracledb change *their* behavior, now it breaks.  We dont want to deal with any of that.  we have a specific contract that we maintain here so "it's just this one line" is really not the full story.

**Comment by dvht** [comment]
OK, thanks, understood.
